docs/01_nodeos/03_plugins/producer_plugin/10_block-producing-explained.md

```diff
@@ -4,87 +4,61 @@ content_title: Block Production Explained
-
-digraph {
-    newrank=true  #allows ranks inside subgraphs (important!)
-    compound=true  #allows edges connecting nodes with subgraphs
-    graph [rankdir=LR]
-    node [style=filled, fillcolor=lightgray, shape=square, fixedsize=true, width=.55, fontsize=10]
-    edge [dir=both, arrowsize=.6, weight=100]
-    splines=false
-
-    subgraph cluster_chain {
-        label="Block Producers Peers"; labelloc="b"
-        graph [color=invis]
-        b0 [label="...", color=invis, style=""]
-        b1 [label="BP-A"]; b2 [label="BP-A\nPeer"]; b3 [label="BP-B\nPeer"]; b4 [label="BP-B"]
-        b5 [label="...", color=invis, style=""]
-        b0 -> b1 -> b2 -> b3 -> b4 -> b5
-    } //cluster_chain
-
-} //digraph
+```
+         +------+     +------+       +------+     +------+
+      -->| BP-A |---->| BP-A |------>| BP-B |---->| BP-B |
+         +------+     | Peer |       | Peer |     +------+
+                      +------+       +------+
 ```
 
-`BP-A` will send block at `p` and,
-
-`BP-B` needs block at time `t` or otherwise will drop it.
+`BP-A` will send block at `l` and, `BP-B` needs block at time `t` or otherwise will drop it.
 
 If `BP-A`is producing 12 blocks as follows `b(lock) at t(ime) 1`, `bt 1.5`, `bt 2`, `bt 2.5`, `bt 3`, `bt 3.5`, `bt 4`, `bt 4.5`, `bt 5`, `bt 5.5`, `bt 6`, `bt 6.5` then `BP-B` needs `bt 6.5` by time `6.5` so it has `.5` to produce `bt 7`.
 
 Please notice that the time of `bt 7` minus `.5` equals the time of `bt 6.5` therefore time `t` is the last block time of `BP-A` and when `BP-B` needs to start its first block.
 
-## Example 1
-`BP-A` has 50% e, m = 200ms, c = 200ms, n = 0ms, a = 250ms:
-`BP-A` sends at (t-250ms) <-> `BP-A-Peer` processes for 200ms and sends at (t - 50ms) <-> `BP-B-Peer` processes for 200ms and sends at (t + 150ms) <-> arrive at `BP-B` 150ms too late.
-
-## Example 2
-`BP-A` has 40% e and m = 200ms, c = 200ms, n = 0ms, a = 300ms:
-(t-300ms) <-> (+200ms) <-> (+200ms) <-> arrive at `BP-B` 100ms too late.
+A block is produced and sent when either it reaches `m` or `n` or `p`. 
 
-## Example 3
-`BP-A` has 30% e and m = 200ms, c = 150ms, n = 0ms, a = 350ms:
-(t-350ms) <-> (+150ms) <-> (+150ms) <-> arrive at `BP-B` with 50ms to spare.
+Starting in Leap 4.0, blocks are propagated after block header validation. This means instead of `BP-A Peer` & `BP-B Peer` taking `m` time to validate and forward a block it only takes a small number of milliseconds to verify the block header and then forward the block.
 
-## Example 4
-`BP-A` has 25% e and m = 200ms, c = 125ms, n = 0ms, a = 375ms:
-(t-375ms) <-> (+125ms) <-> (+125ms) <-> arrive at `BP-B` with 125ms to spare.
+Starting in Leap 5.0, blocks in a round are started immediately after the completion of the previous block. Before 5.0, blocks were always started on `w` intervals and a node would "sleep" between blocks if needed. In 5.0, the "sleeps" are all moved to the end of the block production round. 
```

Comment:
Anytime a block is ready it is sent immediately. That has been true since I think 2.x, not sure which x without digging into it. What is diff in 5.0 is that we start the next block immediately. In previous versions the block would be sent out, but we would wait until the next block boundary to start the next block.

Reply:
That is *not* what the example describes:

* `BP-A` sends b1 at `t1-10ms` => ...

Reply:
The current examples assume non-full blocks. I'll add an example with full blocks.